src/serializer/ResidualHeapSerializer.js

```diff
@@ -1348,14 +1348,10 @@ export class ResidualHeapSerializer {
           to the __empty built-in */
           if (instantRenderMode) {
             if (this.emitter.getReasonToWaitForDependencies(elemVal)) {
-              let error = new CompilerDiagnostic(
+              this.realm.instantRenderBailout(
                 "InstantRender does not yet support cyclical arrays or objects",
```

Comment:
This error is totally unrelated to the previous call to instantRenderBailout and a perfect example of the trap that I want to remove.